.github/workflows/deploy.yml

```
@@ -21,8 +21,52 @@ on:
         type: string
 
 jobs:
+  set-environment-variables:
+    runs-on: ubuntu-latest
+    outputs:
+      stack-name: ${{ steps.construct-stackname.outputs.STACK_NAME }}
+      github-env-name: ${{ steps.construct-stackname.outputs.STACK_NAME }}
+    steps:
+      - name: Checkout
+        id: checkout
+        uses: actions/checkout@v3
+
+      - name: Extract Deployment Variables to Output
+        id: extract-deployment-envs
+        run: |
+          set -o allexport
+          
+          if [ -f "infra/environments/common.env" ]; then
+            source infra/environments/common.env
+          else
+            echo "No common.env file has been found. Disregarding it."
+          fi
+          source infra/environments/${{ inputs.environment_id }}.env
+          
+          if [[ -z $ProductName || -z $AWS_REGION || -z $AWS_ACCOUNT_ID || -z $AWS_ACCOUNT_NAME || -z $AWS_CD_ROLE || -z $AWS_CF_ROLE ]]; then
+            echo -e "All necessary parameters must be defined, which are:\n- ProductName\n- AWS_REGION\n- AWS_ACCOUNT_ID\n- AWS_ACCOUNT_NAME\n- AWS_CD_ROLE\n- AWS_CF_ROLE"
+            exit 1
+          else
+            echo "All necessary parameters are defined"
+          fi
+          
+          echo "ProductName=$ProductName" >> $GITHUB_OUTPUT
+
+      - name: Construct Stack Name
+        id: construct-stackname
+        run: |
+          INPUT_ENV_SUFFIX=${{ inputs.environment_suffix }}
+          if [[ -z $INPUT_ENV_SUFFIX ]]; then
+            echo "STACK_NAME=$( echo ${{ steps.extract-deployment-envs.outputs.ProductName }} )" >> $GITHUB_OUTPUT
+          else
+            echo "STACK_NAME=$( echo ${{ steps.extract-deployment-envs.outputs.ProductName }}-${{ inputs.environment_suffix }} )" >> $GITHUB_OUTPUT
+          fi
```

Do we need the extra `echo` invocations here? Doesn't the following work:
```suggestion
          INPUT_ENV_SUFFIX=${{ inputs.environment_suffix }}
          if [[ -z $INPUT_ENV_SUFFIX ]]; then
            echo "STACK_NAME=${{ steps.extract-deployment-envs.outputs.ProductName }}" >> $GITHUB_OUTPUT
          else
            echo "STACK_NAME=${{ steps.extract-deployment-envs.outputs.ProductName }}-${{ inputs.environment_suffix }}" >> $GITHUB_OUTPUT
          fi
```
(There are a couple more instances with the same pattern in this workflow)

---

Addressed in c77a53c8496ca20210730bdc4240b44780056d2e